torch/_dynamo/source.py

```diff
@@ -178,6 +178,18 @@ def name(self):
         else:
             raise AssertionError(f"unhandled {self.prop}")
 
+    def __eq__(self, other):
+        return (
+            self.base == other.base
+            and self.prop == other.prop
+            and self.idx == other.idx
+        )
```

Comment:
I am confuse; dataclasses automatically get equality defined on them, why you need